sirepo/pkcli/job_agent.py

```diff
@@ -644,19 +645,39 @@ async def _prepare_simulation(self):
                 opName=job.OP_ANALYSIS,
             ),
             send_reply=False,
+            op_id=self.msg.opId,
         )
         await c.start()
         await c._await_exit()
 
     def _sbatch_script(self):
+        def _assert_account():
+            if not self.msg.sbatchRepo:
+                return ''
+            r = subprocess.run(
+                # Info on sed https://stackoverflow.com/a/49430740/5518313
+                # -n == --silent. --silent is not available on BSD
+                "hpssquota | sed -n '/^-\{76\}/,//{//!p;}' | cut -d' ' -f1",
```

Comment:
Don't use shell pipelines in python for parsing:

* There are no guarantees that the pipeline works so you would need to check that. The less complex a shell command the better. 
* Avoids the use of `shell=True` (in this case)
* Testing is harder. You can mock the output of hpssquota and then write a test for the parser.
* It's not Python so it's harder to understand for Pythonistas viz your comment about how to use sed

Also:

* Goal is to validate the repo, not to get a list of repos and check against it so a regex is a better solution
* Before inserting it into a regex, it needs to be validated for specials. I choose a simple regex, which may not be right, but probably is. NERSC doesn't publish syntax.

```py
o = subprocess.check_output(['hpssquota'], text=True)
p = self.msg.sbatchProject
assert re.search(r'^[-\w]+$', p), \
    'invalid NERSC project={r}'
assert re.search(r'{}\s+\d\.'.format(p), o), \
    'sbatchProject={p} is invalid. hpssquota={}'
```py

Reply:
Thanks for the detailed information. For the final regex I think it should be `r'{}\s+\d+\.'` (added a + after \d). This will match `project 123.0`. 
https://github.com/radiasoft/sirepo/pull/2528/commits/ca5b244468a0e5f89041b56fb49a4eac225a8a87

Reply:
Correct